src/ui/popup.js

```diff
@@ -13,16 +11,35 @@ document.querySelectorAll('[data-action]').forEach((element) => {
 
       if (response.ok === false) {
         console.error('Failed to copy message, error: ', response.error);
-        chrome.runtime.sendMessage({ topic: 'badge', params: { type: 'error' } }, () => {
+        chrome.runtime.sendMessage({
+          topic: 'badge',
+          params: { type: 'error' },
+        }, () => {
           window.close();
         });
       }
 
-      copy(response.text).then(() => {
-        chrome.runtime.sendMessage({ topic: 'badge', params: { type: 'success' } }, () => {
-          window.close();
-        });
-      });
+      navigator.clipboard.writeText(response.text)
```

Comment:
In popup.js `navigator.clipboard` will always work because (1) it's a web page and (2) user interaction exists.